base/src/com/google/idea/blaze/base/treeview/BlazeTreeStructureProvider.java

```diff
@@ -63,14 +65,28 @@ public Collection<AbstractTreeNode<?>> modify(
 
     Collection<AbstractTreeNode<?>> result = Lists.newArrayList();
     result.add(rootNode);
+
+    VirtualFile parentVF = rootNode.getVirtualFile() != null ? rootNode.getVirtualFile().getParent() : null;
     for (AbstractTreeNode<?> treeNode : children) {
-      if (keepOriginalNode(treeNode)) {
+      if (keepOriginalNode(treeNode) || isOtherModule(treeNode,parentVF)) {
         result.add(treeNode);
       }
     }
     return result;
   }
 
+  /**
+   * Add modules into project view which has a different ancestor
+   * @param node tree node to be added into project tree view
+   * @param bazelParentVF parent Virtual file of bazel root node
+   * @return true if the tree node's root node is different from the bazel node
+   */
+  private static boolean isOtherModule(AbstractTreeNode<?> node, VirtualFile bazelParentVF){
```

Comment:
what `other` means here? can we have more descriptive name instead?

Reply:
'Other' any module other than modules under the bazel workspace and didn't want to restrict to 'maven'

Reply:
You can name it accordingly: `isOutsideContentRoot` or so. `other` is too generic and your check is quite particular.

Reply:
sure..I will update it with a better one.